libs/ai-endpoints/langchain_nvidia_ai_endpoints/llm.py

```diff
@@ -0,0 +1,223 @@
+from __future__ import annotations
+
+import os
+import warnings
+from typing import Any, Dict, Iterator, List, Optional
+
+from langchain_core.callbacks.manager import CallbackManagerForLLMRun
+from langchain_core.language_models.llms import LLM
+from langchain_core.outputs import GenerationChunk
+from langchain_core.pydantic_v1 import Field, PrivateAttr, root_validator
+
+from langchain_nvidia_ai_endpoints._common import _NVIDIAClient
+from langchain_nvidia_ai_endpoints._statics import Model
+
+
+class NVIDIA(LLM):
+    """
+    LangChain LLM that uses the Completions API with NVIDIA NIMs.
+    """
+
+    class Config:
+        validate_assignment = True
+
+    _client: _NVIDIAClient = PrivateAttr(_NVIDIAClient)
+    _default_model_name: str = "bigcode/starcoder2-7b"
+    _default_base_url: str = "https://integrate.api.nvidia.com/v1"
+    base_url: str = Field(
+        description="Base url for model listing and invocation",
+    )
+    model: Optional[str] = Field(description="The model to use for completions.")
+
+    _base_url_var = "NVIDIA_BASE_URL"
+
+    _init_args: Dict[str, Any] = PrivateAttr()
+    """Stashed arguments given to the constructor that can be passed to
+    the Completions API endpoint."""
+
+    @root_validator(pre=True)
+    def _validate_base_url(cls, values: Dict[str, Any]) -> Dict[str, Any]:
```

Comment:
we already have validation logic in client right? can this be moved to _common? I see repeating code in reranking and embedding class too